### PR TITLE
[Merged by Bors] - feat: add a discrete topology instance for zlattice

### PR DESCRIPTION
**Pull request by xroblot**
Add the following
```lean
variable [NormedAddCommGroup E] [NormedSpace ℝ E] (b : Basis ι ℝ E)

instance [Fintype ι] : DiscreteTopology (span ℤ (Set.range b)) 
```

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Zlattice.lean [hunk 1]
```diff
@@ -288,9 +288,29 @@ end NormedLatticeField
...
+  · rw [← hy, ← Int.cast_abs, ← Int.cast_one,  Int.cast_lt, Int.abs_lt_one_iff, Int.cast_eq_zero]
+  exact ((Pi.basisFun ℝ ι).mem_span_iff_repr_mem ℤ x).mp (SetLike.coe_mem x) i
+
+variable [NormedAddCommGroup E] [NormedSpace ℝ E] (b : Basis ι ℝ E)
+
+instance [Fintype ι] : DiscreteTopology (span ℤ (Set.range b)) := by
+  have h : Set.MapsTo b.equivFun (span ℤ (Set.range b)) (span ℤ (Set.range (Pi.basisFun ℝ ι))) := by
+    intro _ hx
+    rw [SetLike.mem_coe, Basis.mem_span_iff_repr_mem] at hx ⊢
+    exact fun i => hx i
```
Comment (alreadydone) [inline Mathlib/Algebra/Module/Zlattice.lean:310]:
```suggestion
    rwa [SetLike.mem_coe, Basis.mem_span_iff_repr_mem] at hx ⊢
```

File: Mathlib/Algebra/Module/Zlattice.lean [hunk 2]
```diff
@@ -288,9 +288,29 @@ end NormedLatticeField
...
-variable (b : Basis ι ℝ E)
+theorem discreteTopology_pi_basisFun [Fintype ι] :
+    DiscreteTopology (span ℤ (Set.range (Pi.basisFun ℝ ι))) := by
+  refine discreteTopology_iff_open_singleton_zero.mpr ⟨Metric.ball 0 1, Metric.isOpen_ball, ?_⟩
+  ext x
+  rw [Set.mem_preimage, mem_ball_zero_iff, Set.mem_singleton_iff, Pi.norm_def,
+    ← Real.lt_toNNReal_iff_coe_lt, Finset.sup_lt_iff (by simp), Real.toNNReal_one, ← coe_eq_zero]
+  simp_rw [Finset.mem_univ, forall_true_left, Function.funext_iff, Pi.zero_apply]
+  refine forall_congr' (fun i => ?_)
+  rw [← norm_toNNReal, Real.toNNReal_lt_one, Real.norm_eq_abs]
```
Comment (alreadydone) [inline Mathlib/Algebra/Module/Zlattice.lean:299]:
```suggestion
  rw [Set.mem_preimage, mem_ball_zero_iff, pi_norm_lt_iff zero_lt_one, Set.mem_singleton_iff]
  simp_rw [← coe_eq_zero, Function.funext_iff, Pi.zero_apply, Real.norm_eq_abs]
  refine forall_congr' (fun i => ?_)
```

**Review by alreadydone — APPROVED** [review]

File: Mathlib/Algebra/Module/Zlattice.lean (hunk 2 above)
Comment (xroblot) [reply Mathlib/Algebra/Module/Zlattice.lean:299]: Thanks!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**